components/script/dom/formdata.rs

```diff
@@ -163,4 +164,35 @@ impl FormData {
 
         ret
     }
+
+    pub fn atoms(&self) -> Vec<Atom> {
+        let mut ret = vec![];
+        for (key, value) in self.data.borrow().iter() {
```

Comment:
This is an anti-pattern, I'm pretty sure. You can use [Iterators](https://doc.rust-lang.org/std/iter/trait.Iterator.html) to do lots of amazing things.